packages/react/src/lib/hooks/useFetch/README.md

```diff
@@ -0,0 +1,114 @@
+## useFetch
+
+This hook can be used on conjunction with other backend services configured to work with Nile. All it does is ensure `credentials` are set to `include` to be sure cookies HTTP-only cookies are passed to the secondary service.
+
+### Usage
+
+```typescript
+import { Stack, Button } from '@mui/joy';
+import { useNile, useFetch } from '@thenile.dev/react';
+
+function CustomComponent() {
+  const nile = useNile();
+  const fetch = useFetch();
+
+  const fetchFromBackend = React.useCallback(() => {
+    async function doRequest() {
+      try {
+        const myResult = await fetch('https://my.backend.service');
+        console.log(myResult);
+      } catch (e) {
+        console.error(e.message);
+      }
+    }
+    doRequest();
+  }, []);
+
+  const fetchFromNile = React.useCallback(() => {
+    async function doRequest() {
+      try {
+        const myResult = await nile.users.listUsers();
+        console.log(myResult);
+      } catch (e) {
+        console.error(e.message);
+      }
+    }
+    doRequest();
+  }, []);
+
+  return (
+    <Stack>
+      <Button onClick={fetchFromNile}>Do Request from Nile</Button>;
+      <Button onClick={fetchFromBackend}>Do Request from backend</Button>;
+    </Stack>
+  );
+}
+```
+
+## Handling CORS
+
+In order for `useFetch` and `useNile` to work well together, the secondary web server needs set up to handle CORS. This allows for cookies to be shared transparently between your domain and Nile. Below is an example (in nextjs) for using Nile as the source of truth for authentication on a secondary server:
+
+```typescript
+import type { NextApiRequest, NextApiResponse } from 'next';
+import Cors from 'cors';
+import Nile from '@theniledev/js';
+
+// Initialize Nile
+const nile = Nile({
+  basePath: 'https://prod.thenile.dev',
+  workspace: 'myWorkspace',
+});
+
+// Initializing the cors middleware
+// You can read more about the available options here: https://github.com/expressjs/cors#configuration-options
+const cors = Cors({
+  origin: ['https://mysupercoolsite.com'], // **important** the `origin` here needs to match the `origin` CORS header on `https://prod.thenile.dev`
```

Comment:
How does one determine what is the `origin` CORS header on `https://prod.thenile.dev`?

Reply:
:magic: :homer_disappear:

you'd have to look at the request headers... and you can't change the value right now, so this is extremely handwavey.

Reply:
FYC add a line that says that (inspect the headers...etc)

Reply:
in currently instructs you to match the CORS header.